go/adbc/driver/snowflake/snowflake_database.go

```diff
@@ -209,7 +209,7 @@ func (d *databaseImpl) SetOptionInternal(k string, v string, cnOptions *map[stri
 	case OptionRegion:
 		d.cfg.Region = v
 	case OptionAccount:
-		d.cfg.Account = v
+		d.cfg.Account = strings.ReplaceAll(v, "_", "-")
```

Comment:
I'm pretty sure this is better than putting it in xdbc because xdbc would have to start making control flow for what to do depending on the backend in use and that seems like a mess.

Reply:
A comment with links explaining this would be very nice to have here.

Reply:
Wait, what the heck, there should be a very nice comment

Reply:
Oh it got reverted out when I cleared out some formatting commits.

Reply:
There you go. I did all that research into the RFCs I'm leaving a link 😆